src/rgw/motr/sync/motr_sync.h

```diff
@@ -0,0 +1,99 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=2 sw=2 expandtab ft=cpp
+
+/*
+ * Abstract thread/process synchronization for the CORTX Motr backend
+ *
+ * Copyright (C) 2022 Seagate Technology LLC and/or its Affiliates
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ *
+ */
+
+#ifndef __MOTR_SYNCHROZATION__
+#define __MOTR_SYNCHROZATION__
+
+#include <unistd.h>
+
+#include "include/types.h"
+#include "include/utime.h"
+#include "rgw_sal_motr.h"
+
+// Define MOTR_EXCLUSIVE_RW_LOCK only when system needs exclusive read/write lock
+#define MOTR_EXCLUSIVE_RW_LOCK 1
+
+class MotrLockProvider;
+
+struct motr_locker_info_t
+{
+  std::string coockie;        // unique id of caller
```

Comment:
What is `coockie`?

Reply:
coockie is identifier of an lock holder, we can call it `lock_holder` or `holder_id` as well. 
but to keep it generic @DPG17 has kept it as `cookie` I think.

Reply:
Yes, it should be `cookie` and not `coockie`

Reply:
done